test/e2e/instrumentation/monitoring/metrics_grabber.go

```diff
@@ -17,61 +17,54 @@ limitations under the License.
 package monitoring
 
 import (
-	"context"
+	"errors"
 	"fmt"
-	"strings"
 	"time"
 
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clientset "k8s.io/client-go/kubernetes"
 
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2emetrics "k8s.io/kubernetes/test/e2e/framework/metrics"
 	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 	instrumentation "k8s.io/kubernetes/test/e2e/instrumentation/common"
 )
 
 var _ = instrumentation.SIGDescribe("MetricsGrabber", func() {
 	f := framework.NewDefaultFramework("metrics-grabber")
 	var c, ec clientset.Interface
 	var grabber *e2emetrics.Grabber
-	var masterRegistered bool
 	ginkgo.BeforeEach(func() {
 		var err error
 		c = f.ClientSet
 		ec = f.KubemarkExternalClusterClientSet
-		// Check if master Node is registered
-		nodes, err := c.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
-		framework.ExpectNoError(err)
-		for _, node := range nodes.Items {
-			if strings.HasSuffix(node.Name, "master") {
-				masterRegistered = true
-			}
-		}
 		gomega.Eventually(func() error {
-			grabber, err = e2emetrics.NewMetricsGrabber(c, ec, true, true, true, true, true, true)
+			grabber, err = e2emetrics.NewMetricsGrabber(c, ec, f.ClientConfig(), true, true, true, true, true, true)
 			if err != nil {
 				return fmt.Errorf("failed to create metrics grabber: %v", err)
 			}
-			if masterRegistered && !grabber.HasControlPlanePods() {
-				return fmt.Errorf("unable to get find control plane pods")
-			}
 			return nil
 		}, 5*time.Minute, 10*time.Second).Should(gomega.BeNil())
 	})
 
 	ginkgo.It("should grab all metrics from API server.", func() {
 		ginkgo.By("Connecting to /metrics endpoint")
 		response, err := grabber.GrabFromAPIServer()
+		if errors.Is(err, e2emetrics.MetricsGrabbingDisabledError) {
+			e2eskipper.Skipf("%v", err)
+		}
 		framework.ExpectNoError(err)
 		gomega.Expect(response).NotTo(gomega.BeEmpty())
```

Comment:
can we log the response in all tests?
 currently is not clear from the logs if the test passes or not

Reply:
Where do you want to add logging? In the caller of `e2eskipper.Skipf` or in `Skipf` itself?

Not being able to determine that and why a test was skipped looks like a general problem. I remember that JUnit records the skip reason, but prow.k8s.io doesn't show that. Therefore I am leaning towards logging in `Skipf` as a band-aid.

Reply:
I was thinking just to log the whole response
```
framework.Logf("Returned metrics: %v", response)
```
The skip can have a more complete message as in:
```
e2eskipper.Skipf("Skipping test, can't grab metrics: %v", err)
```

Reply:
>  I remember that JUnit records the skip reason, but prow.k8s.io doesn't show that.

If there is junit data that is not possible to view on prow.k8s.io please file a test-infra issue so we can see if someone can fix this.

Reply:
> I was thinking just to log the whole response

Isn't that going to be a bit verbose? But I can add it.

> The skip can have a more complete message as in:
> `e2eskipper.Skipf("Skipping test, can't grab metrics: %v", err)`

That makes less sense to me:
- It's a skip message, explicitly adding "Skipping test" is redundant. Other calls to `Skipf` don't have it.
- `can't grab metrics` is redundant because the error already contains `metrics grabbing disabled`

> If there is junit data that is not possible to view on prow.k8s.io please file a test-infra issue so we can see if someone can fix this.

The JUnit file must have been from some other test tool, the junit files from e2e.test don't contain the skip reason. That makes it an issue for k/k: https://github.com/kubernetes/kubernetes/issues/103430

Whether prow.io then would show the skip reason remains to be seen.

Reply:
feel free to scratch those comments, they are very hacky

Reply:
hmm I think that the problem is that the test is marked as PASSED instead of SKIPPED

Reply:
sorry, last day before holidays, I'm not thinking straight today, it does show as skipped

Reply:
It should be marked skipped when skipped, if it's not marked skipped it could only be that some conditional was used internally. I don't think we need to log skipping specially here but it would certainly be nice if we recorded why tests were skipped more generally